<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Source Themes Academic 4.7.0"><meta name=author content="Jiannan Zhang"><meta name=description content="为什么要在R中进行ID转换 在转录组数据或者其他分析中，好吧，其实是因为我做的物种是家鸡，根本就没有人和鼠那样有着非常完善和时刻更新的数据库做"><link rel=alternate hreflang=en-us href=/post/gene-id%E8%BD%AC%E6%8D%A2/><meta name=theme-color content="#4caf50"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/styles/github.min.css crossorigin=anonymous title=hl-light><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/styles/dracula.min.css crossorigin=anonymous title=hl-dark disabled><script src=https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Lato:400,700%7CNoto Serif SC:400,700,800%7CRoboto+Mono&display=swap"><link rel=stylesheet href=/css/academic.css><script async src="https://www.googletagmanager.com/gtag/js?id=UA-162031575-1"></script><script>window.dataLayer=window.dataLayer||[];function gtag(){dataLayer.push(arguments);}
function trackOutboundLink(url,target){gtag('event','click',{'event_category':'outbound','event_label':url,'transport_type':'beacon','event_callback':function(){if(target!=='_blank'){document.location=url;}}});console.debug("Outbound link clicked: "+url);}
function onClickCallback(event){if((event.target.tagName!=='A')||(event.target.host===window.location.host)){return;}
trackOutboundLink(event.target,event.target.getAttribute('target'));}
gtag('js',new Date());gtag('config','UA-162031575-1',{});document.addEventListener('click',onClickCallback,false);</script><link rel=manifest href=/index.webmanifest><link rel=icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_2.png><link rel=apple-touch-icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_2.png><link rel=canonical href=/post/gene-id%E8%BD%AC%E6%8D%A2/><meta property="twitter:card" content="summary"><meta property="twitter:site" content="@Jiannan_scu"><meta property="twitter:creator" content="@Jiannan_scu"><meta property="og:site_name" content="Jiannan Zhang"><meta property="og:url" content="/post/gene-id%E8%BD%AC%E6%8D%A2/"><meta property="og:title" content="Gene ID转换 | Jiannan Zhang"><meta property="og:description" content="为什么要在R中进行ID转换 在转录组数据或者其他分析中，好吧，其实是因为我做的物种是家鸡，根本就没有人和鼠那样有着非常完善和时刻更新的数据库做"><meta property="og:image" content="img/map[gravatar:%!s(bool=false) shape:circle]"><meta property="twitter:image" content="img/map[gravatar:%!s(bool=false) shape:circle]"><meta property="og:locale" content="en-us"><meta property="article:published_time" content="2020-03-13T00:00:00+00:00"><meta property="article:modified_time" content="2020-03-13T13:54:04+08:00"><script type=application/ld+json>{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"/post/gene-id%E8%BD%AC%E6%8D%A2/"},"headline":"Gene ID转换","datePublished":"2020-03-13T00:00:00Z","dateModified":"2020-03-13T13:54:04+08:00","author":{"@type":"Person","name":"Jiannan Zhang"},"publisher":{"@type":"Organization","name":"Jiannan Zhang","logo":{"@type":"ImageObject","url":"/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_2.png"}},"description":"为什么要在R中进行ID转换 在转录组数据或者其他分析中，好吧，其实是因为我做的物种是家鸡，根本就没有人和鼠那样有着非常完善和时刻更新的数据库做"}</script><title>Gene ID转换 | Jiannan Zhang</title></head><body id=top data-spy=scroll data-offset=70 data-target=#TableOfContents><aside class=search-results id=search><div class=container><section class=search-header><div class="row no-gutters justify-content-between mb-3"><div class=col-6><h1>Search</h1></div><div class="col-6 col-search-close"><a class=js-search href=#><i class="fas fa-times-circle text-muted" aria-hidden=true></i></a></div></div><div id=search-box><input name=q id=search-query placeholder=Search... autocapitalize=off autocomplete=off autocorrect=off spellcheck=false type=search></div></section><section class=section-search-results><div id=search-hits></div></section></div></aside><nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main><div class=container><div class="d-none d-lg-inline-flex"><a class=navbar-brand href=/>Jiannan Zhang</a></div><button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar aria-expanded=false aria-label="Toggle navigation">
<span><i class="fas fa-bars"></i></span></button><div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none"><a class=navbar-brand href=/>Jiannan Zhang</a></div><div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content><ul class="navbar-nav d-md-inline-flex"><li class=nav-item><a class=nav-link href=/#posts><span>Posts</span></a></li><li class=nav-item><a class=nav-link href=/#featured><span>Publications</span></a></li><li class=nav-item><a class=nav-link href=/#projects><span>Projects</span></a></li><li class=nav-item><a class=nav-link href=/courses/><span>Courses</span></a></li><li class=nav-item><a class=nav-link href=/#contact><span>Contact</span></a></li></ul></div><ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2"><li class=nav-item><a class="nav-link js-search" href=#><i class="fas fa-search" aria-hidden=true></i></a></li><li class=nav-item><a class="nav-link js-dark-toggle" href=#><i class="fas fa-moon" aria-hidden=true></i></a></li></ul></div></nav><article class=article><div class="article-container pt-3"><h1>Gene ID转换</h1><div class=article-metadata><span class=article-date>Last updated on
Mar 13, 2020</span>
<span class=middot-divider></span><span class=article-reading-time>5 min read</span>
<span class=middot-divider></span><span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/categories/r/>R</a></span></div></div><div class=article-container><div class=article-style><script src=/rmarkdown-libs/header-attrs/header-attrs.js></script><div id=为什么要在r中进行id转换 class="section level3"><h3>为什么要在R中进行ID转换</h3><p>在转录组数据或者其他分析中，好吧，其实是因为我做的物种是家鸡，根本就没有人和鼠那样有着非常完善和时刻更新的数据库做为支持，我经常会遇见GeneID转换，同源基因转换的问题，之前一直利用<code>Ensembl</code>主页里面的<code>biomart</code>进行数据下载，然后在excel里面利用vlookup进行操作，哎，费时费力，而且不能高度自动化和重复化<del>（传统湿实验室人员的基本操作思维）</del>，现在已经开始将大部分操作都在R里面进行了，刚好学习了<code>biomaRt</code>包可以完美替代之前的操作流程，下面就是相关笔记及备注：</p><blockquote><p>好吧，这是第一篇Rmd笔记，Rmarkdown真的超级好用啊！</p></blockquote></div><div id=安装所需的软件包 class="section level3"><h3>安装所需的软件包</h3><p>下面的R包也包括Y叔的<code>clusterProfiler</code>了，里面也有ID转换的板块，也是非常实用的，一并汇总了。</p><pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
  install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;biomaRt&quot;)</code></pre></div><div id=biomart示例 class="section level3"><h3><code>biomaRt</code>示例</h3><p>首先还是推荐看<a href=https://www.bioconductor.org/packages/devel/bioc/vignettes/biomaRt/inst/doc/biomaRt.html>官方文档</a>，没有什么攻略比官方文档更靠谱了，除非懒~</p><pre class=r><code>library(&quot;biomaRt&quot;)
library(org.Gg.eg.db)
library(org.Hs.eg.db)
library(clusterProfiler)
library(DOSE)

# 查看Marts库
listMarts(host=&quot;asia.ensembl.org&quot;)</code></pre><pre><code>##                biomart               version
## 1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 99
## 2   ENSEMBL_MART_MOUSE      Mouse strains 99
## 3     ENSEMBL_MART_SNP  Ensembl Variation 99
## 4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 99</code></pre></div><div id=一步到位代码 class="section level3"><h3>一步到位代码</h3><p>选择所需的数据库代码，我需要的是人和家鸡的数据库，直接定义到<code>dataset</code>即可，有时网速很慢，添加了对应地区的<code>host</code>。</p><pre class=r><code># 定义人对应数据mart
human = useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;,host=&quot;asia.ensembl.org&quot;)

# 定义家鸡对应数据mart
chicken = useMart(&quot;ensembl&quot;,dataset=&quot;ggallus_gene_ensembl&quot;,host=&quot;asia.ensembl.org&quot;)

# 查看对应`attributes`说明
listAttributes(chicken)[1:40,]</code></pre><pre><code>##                               name                                description
## 1                  ensembl_gene_id                             Gene stable ID
## 2          ensembl_gene_id_version                     Gene stable ID version
## 3            ensembl_transcript_id                       Transcript stable ID
## 4    ensembl_transcript_id_version               Transcript stable ID version
## 5               ensembl_peptide_id                          Protein stable ID
## 6       ensembl_peptide_id_version                  Protein stable ID version
## 7                  ensembl_exon_id                             Exon stable ID
## 8                      description                           Gene description
## 9                  chromosome_name                   Chromosome/scaffold name
## 10                  start_position                            Gene start (bp)
## 11                    end_position                              Gene end (bp)
## 12                          strand                                     Strand
## 13                            band                             Karyotype band
## 14                transcript_start                      Transcript start (bp)
## 15                  transcript_end                        Transcript end (bp)
## 16        transcription_start_site             Transcription start site (TSS)
## 17               transcript_length Transcript length (including UTRs and CDS)
## 18               transcript_appris                          APPRIS annotation
## 19              external_gene_name                                  Gene name
## 20            external_gene_source                        Source of gene name
## 21        external_transcript_name                            Transcript name
## 22 external_transcript_source_name                  Source of transcript name
## 23                transcript_count                           Transcript count
## 24      percentage_gene_gc_content                          Gene % GC content
## 25                    gene_biotype                                  Gene type
## 26              transcript_biotype                            Transcript type
## 27                          source                              Source (gene)
## 28               transcript_source                        Source (transcript)
## 29                         version                             Version (gene)
## 30              transcript_version                       Version (transcript)
## 31                 peptide_version                          Version (protein)
## 32                external_synonym                               Gene Synonym
## 33           phenotype_description                      Phenotype description
## 34                     source_name                                Source name
## 35               study_external_id                   Study external reference
## 36                     strain_name                                Strain name
## 37                   strain_gender                              Strain gender
## 38                         p_value                                    P value
## 39                           go_id                          GO term accession
## 40                       name_1006                               GO term name
##            page
## 1  feature_page
## 2  feature_page
## 3  feature_page
## 4  feature_page
## 5  feature_page
## 6  feature_page
## 7  feature_page
## 8  feature_page
## 9  feature_page
## 10 feature_page
## 11 feature_page
## 12 feature_page
## 13 feature_page
## 14 feature_page
## 15 feature_page
## 16 feature_page
## 17 feature_page
## 18 feature_page
## 19 feature_page
## 20 feature_page
## 21 feature_page
## 22 feature_page
## 23 feature_page
## 24 feature_page
## 25 feature_page
## 26 feature_page
## 27 feature_page
## 28 feature_page
## 29 feature_page
## 30 feature_page
## 31 feature_page
## 32 feature_page
## 33 feature_page
## 34 feature_page
## 35 feature_page
## 36 feature_page
## 37 feature_page
## 38 feature_page
## 39 feature_page
## 40 feature_page</code></pre><pre class=r><code>listAttributes(human)[1:10,]</code></pre><pre><code>##                             name                  description         page
## 1                ensembl_gene_id               Gene stable ID feature_page
## 2        ensembl_gene_id_version       Gene stable ID version feature_page
## 3          ensembl_transcript_id         Transcript stable ID feature_page
## 4  ensembl_transcript_id_version Transcript stable ID version feature_page
## 5             ensembl_peptide_id            Protein stable ID feature_page
## 6     ensembl_peptide_id_version    Protein stable ID version feature_page
## 7                ensembl_exon_id               Exon stable ID feature_page
## 8                    description             Gene description feature_page
## 9                chromosome_name     Chromosome/scaffold name feature_page
## 10                start_position              Gene start (bp) feature_page</code></pre><pre class=r><code># 进行ID转换

hgnc_swissprot &lt;- getBM(attributes=c(&#39;ensembl_gene_id&#39;,&#39;ensembl_transcript_id&#39;,&#39;hgnc_symbol&#39;,&#39;uniprotswissprot&#39;),filters = &#39;ensembl_gene_id&#39;, values = &#39;ENSG00000139618&#39;, mart = human)

hgnc_swissprot</code></pre><pre><code>##   ensembl_gene_id ensembl_transcript_id hgnc_symbol uniprotswissprot
## 1 ENSG00000139618       ENST00000671466       BRCA2                 
## 2 ENSG00000139618       ENST00000380152       BRCA2           P51587
## 3 ENSG00000139618       ENST00000530893       BRCA2                 
## 4 ENSG00000139618       ENST00000670614       BRCA2                 
## 5 ENSG00000139618       ENST00000665585       BRCA2                 
## 6 ENSG00000139618       ENST00000528762       BRCA2                 
## 7 ENSG00000139618       ENST00000470094       BRCA2                 
## 8 ENSG00000139618       ENST00000666593       BRCA2                 
## 9 ENSG00000139618       ENST00000544455       BRCA2           P51587</code></pre></div><div id=同源基因id转换代码 class="section level3"><h3>同源基因ID转换代码</h3><p>跨物种同源基因转换使用<code>getLDS</code>命令即可，也是非常方便，但是有一个问题就是家鸡一个基因可能对应多个人的基因，系统会自动添加对应的旁系同源基因，所以转换后的数据使用需要小心，可能需要人工排查一下，或者人为的去掉部分数据。</p><pre class=r><code>#human = useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;,host=&quot;asia.ensembl.org&quot;)
#chicken = useMart(&quot;ensembl&quot;,dataset=&quot;ggallus_gene_ensembl&quot;,host=&quot;asia.ensembl.org&quot;)

genelist_megre &lt;- c(&quot;ENSGALG00000001136&quot;,&quot;ENSGALG00000005956&quot;,&quot;ENSGALG00000006407&quot;,&quot;ENSGALG00000007728&quot;,&quot;ENSGALG00000009628&quot;,&quot;ENSGALG00000010316&quot;,&quot;ENSGALG00000011190&quot;,&quot;ENSGALG00000011687&quot;,&quot;ENSGALG00000011806&quot;,&quot;ENSGALG00000015656&quot;)

genelist_megre</code></pre><pre><code>##  [1] &quot;ENSGALG00000001136&quot; &quot;ENSGALG00000005956&quot; &quot;ENSGALG00000006407&quot;
##  [4] &quot;ENSGALG00000007728&quot; &quot;ENSGALG00000009628&quot; &quot;ENSGALG00000010316&quot;
##  [7] &quot;ENSGALG00000011190&quot; &quot;ENSGALG00000011687&quot; &quot;ENSGALG00000011806&quot;
## [10] &quot;ENSGALG00000015656&quot;</code></pre><pre class=r><code>genelist_megre_Hs &lt;- getLDS(attributes = c(&quot;ensembl_gene_id&quot;,&quot;entrezgene_id&quot;, &quot;external_gene_name&quot;),      
                            filters = &quot;ensembl_gene_id&quot;, 
                            values = genelist_megre, 
                            mart = chicken,
                            attributesL = c(&quot;ensembl_gene_id&quot;,&quot;entrezgene_id&quot;,&quot;external_gene_name&quot;), 
                            martL = human)

genelist_megre_Hs</code></pre><pre><code>##        Gene.stable.ID NCBI.gene.ID Gene.name Gene.stable.ID.1 NCBI.gene.ID.1
## 1  ENSGALG00000007728       770734            ENSG00000100033           5625
## 2  ENSGALG00000005956       423774            ENSG00000264230         728113
## 3  ENSGALG00000011687    100859120            ENSG00000185567         113146
## 4  ENSGALG00000005956       423774            ENSG00000265190         653145
## 5  ENSGALG00000011190    100857411     PLAC8  ENSG00000145287          51316
## 6  ENSGALG00000011806    100859223            ENSG00000173702          56667
## 7  ENSGALG00000006407       378902            ENSG00000120889           8795
## 8  ENSGALG00000006407       378902            ENSG00000173535           8794
## 9  ENSGALG00000006407       378902            ENSG00000173530           8793
## 10 ENSGALG00000007728       770734            ENSG00000277196      102724788
##    Gene.name.1
## 1        PRODH
## 2      ANXA8L1
## 3       AHNAK2
## 4        ANXA8
## 5        PLAC8
## 6        MUC13
## 7    TNFRSF10B
## 8    TNFRSF10C
## 9    TNFRSF10D
## 10  AC007325.2</code></pre><blockquote><p>插一句，总算知道为什么R代码有规范了，参数主动换行，真的好看太多了。</p></blockquote></div><div id=提取蛋白dna序列并导出fasta格式文件 class="section level3"><h3>提取蛋白/DNA序列并导出FASTA格式文件</h3><p>使用过程中发现居然还可以调取序列等信息，少量操作也就不用打开网页，快速方便完成氨基酸/核酸序列的获取，非常实用！</p><pre class=r><code>protein = getSequence(id=c(&quot;GRP&quot;),
                      type=&quot;external_gene_name&quot;,
                      seqType=&quot;peptide&quot;, 
                      mart= chicken)

protein</code></pre><pre><code>##                                                                                                                                                                 peptide
## 1              MGGGGPRRPGTLPLLALLALLAAHGGAAPLQPGGSPALTKIYPRGSHWAVGHLMGKKSTGDFPYAYEEENKIPLSASPENIKQLDDYLQREEMSKHLLQLLEGNENKSAHFSKGGLPWHTRNSWETDDSSSWKDVVEYLLQVVNMKESAPS*
## 2 MGGGGPRRPGTLPLLALLALLAAHGGAAPLQPGGSPALTKIYPRGSHWAVGHLMGKKSTGDFPYAYEEENKIPLSASPENIKQLDDYLQREEMSKHLLQLLEGNENKSAHFSKGGLPWHTRNSWETDDSSSWKDVSRTRCVSAFLTVTFCSKVAYQLCPTSALS*
##   external_gene_name
## 1                GRP
## 2                GRP</code></pre></div><div id=混杂基因名称转换处理这里使用了bitr进行示例 class="section level3"><h3>混杂基因名称转换处理，这里使用了<code>bitr</code>进行示例</h3><p>在获得基因列表是不可避免会出现name和ID混杂在一起的情况，目前只能采用分批操作的流程进行处理，有更好方法再更新。</p><pre class=r><code>genelist &lt;- c(&quot;S100A6&quot;,&quot;S100A10&quot;,&quot;SNCG&quot;,&quot;S100A1&quot;,&quot;S100A11&quot;,&quot;ID3&quot;,&quot;ENSGALG00000011190&quot;,&quot;AKR1B10&quot;,&quot;CSRP1&quot;,&quot;SPARCL1&quot;,&quot;HBEGF&quot;,&quot;FAM107B&quot;,&quot;ENSGALG00000009628&quot;,&quot;ENSGALG00000010316&quot;,&quot;ENSGALG00000011190&quot;,&quot;ENSGALG00000011687&quot;,&quot;ENSGALG00000011806&quot;,&quot;ENSGALG00000015656&quot;)

genelist</code></pre><pre><code>##  [1] &quot;S100A6&quot;             &quot;S100A10&quot;            &quot;SNCG&quot;              
##  [4] &quot;S100A1&quot;             &quot;S100A11&quot;            &quot;ID3&quot;               
##  [7] &quot;ENSGALG00000011190&quot; &quot;AKR1B10&quot;            &quot;CSRP1&quot;             
## [10] &quot;SPARCL1&quot;            &quot;HBEGF&quot;              &quot;FAM107B&quot;           
## [13] &quot;ENSGALG00000009628&quot; &quot;ENSGALG00000010316&quot; &quot;ENSGALG00000011190&quot;
## [16] &quot;ENSGALG00000011687&quot; &quot;ENSGALG00000011806&quot; &quot;ENSGALG00000015656&quot;</code></pre><pre class=r><code># 定义ENSEMBL命名的基因
index &lt;- grep(&quot;ENSGALG&quot;, genelist)

index</code></pre><pre><code>## [1]  7 13 14 15 16 17 18</code></pre><pre class=r><code># 筛选ENSGALG命名的基因并转换
genelistA &lt;- genelist[index]
geneA = bitr(genelistA, fromType=&quot;ENSEMBL&quot;, toType=c(&quot;SYMBOL&quot;, &quot;ENTREZID&quot;), OrgDb=&quot;org.Gg.eg.db&quot;)
head(geneA)</code></pre><pre><code>##              ENSEMBL SYMBOL  ENTREZID
## 1 ENSGALG00000011190 PLACL2 100857411
## 2 ENSGALG00000009628 KCTD12    425504
## 3 ENSGALG00000010316  FRAS1    422506
## 4 ENSGALG00000011687 AHNAK2 100859120
## 5 ENSGALG00000011806  MUC13 100859223
## 6 ENSGALG00000015656  AKAP2 100533110</code></pre><pre class=r><code># 筛选SYMBOL命名的基因并转换
genelistB &lt;- genelist[-index]
geneB = bitr(genelistB, fromType=&quot;SYMBOL&quot;, toType=c(&quot;ENSEMBL&quot;, &quot;ENTREZID&quot;), OrgDb=&quot;org.Gg.eg.db&quot;)</code></pre><pre><code>## Warning in bitr(genelistB, fromType = &quot;SYMBOL&quot;, toType = c(&quot;ENSEMBL&quot;,
## &quot;ENTREZID&quot;), : 9.09% of input gene IDs are fail to map...</code></pre><pre class=r><code>head(geneB)</code></pre><pre><code>##    SYMBOL            ENSEMBL  ENTREZID
## 1  S100A6 ENSGALG00000041826    373951
## 2 S100A10 ENSGALG00000028774    396506
## 3    SNCG ENSGALG00000002015    395392
## 4  S100A1 ENSGALG00000034458 100858258
## 5 S100A11 ENSGALG00000036711    396075
## 6     ID3 ENSGALG00000035317    395281</code></pre><pre class=r><code>gene &lt;- rbind(geneA,geneB)

gene</code></pre><pre><code>##               ENSEMBL  SYMBOL  ENTREZID
## 1  ENSGALG00000011190  PLACL2 100857411
## 2  ENSGALG00000009628  KCTD12    425504
## 3  ENSGALG00000010316   FRAS1    422506
## 4  ENSGALG00000011687  AHNAK2 100859120
## 5  ENSGALG00000011806   MUC13 100859223
## 6  ENSGALG00000015656   AKAP2 100533110
## 12 ENSGALG00000041826  S100A6    373951
## 21 ENSGALG00000028774 S100A10    396506
## 31 ENSGALG00000002015    SNCG    395392
## 41 ENSGALG00000034458  S100A1 100858258
## 51 ENSGALG00000036711 S100A11    396075
## 61 ENSGALG00000035317     ID3    395281
## 8  ENSGALG00000000318   CSRP1    396176
## 9  ENSGALG00000010929 SPARCL1    422586
## 10 ENSGALG00000000949   HBEGF    395654
## 11 ENSGALG00000033226 FAM107B    426376</code></pre><pre class=r><code>summary(gene)</code></pre><pre><code>##    ENSEMBL             SYMBOL            ENTREZID        
##  Length:16          Length:16          Length:16         
##  Class :character   Class :character   Class :character  
##  Mode  :character   Mode  :character   Mode  :character</code></pre><pre class=r><code># 同时可以导出到本地csv文件
#write.csv(as.data.frame(gene),&quot;./gene.csv&quot;,row.names =FALSE)</code></pre></div><div id=当然使用getbm也可以实现id转换但是结果会出现不少信息残缺还是推荐bitr class="section level3"><h3>当然，使用<code>getBM</code>也可以实现ID转换，但是结果会出现不少信息残缺，还是推荐<code>bitr</code></h3><pre class=r><code>gene_A &lt;- getBM(attributes=c(&quot;ensembl_gene_id&quot;, &quot;external_gene_name&quot;, &quot;entrezgene_id&quot;),
                filters = &#39;ensembl_gene_id&#39;, 
                values = genelistA, 
                mart = chicken)

gene_B &lt;- getBM(attributes=c(&quot;ensembl_gene_id&quot;, &quot;external_gene_name&quot;, &quot;entrezgene_id&quot;),
                filters = &#39;external_gene_name&#39;, 
                values = genelistB, 
                mart = chicken)

gene_ABmegre &lt;- rbind(gene_A,gene_B)
gene_ABmegre</code></pre><pre><code>##       ensembl_gene_id external_gene_name entrezgene_id
## 1  ENSGALG00000009628                           425504
## 2  ENSGALG00000010316                           422506
## 3  ENSGALG00000011190              PLAC8     100857411
## 4  ENSGALG00000011687                        100859120
## 5  ENSGALG00000011806                        100859223
## 6  ENSGALG00000015656         PALM2AKAP2     100533110
## 7  ENSGALG00000028774            S100A10        396506
## 8  ENSGALG00000000318              CSRP1        396176
## 9  ENSGALG00000036711            S100A11        396075
## 10 ENSGALG00000000949              HBEGF        395654
## 11 ENSGALG00000002015               SNCG        395392
## 12 ENSGALG00000013072            AKR1B10        395338
## 13 ENSGALG00000041826             S100A6        373951
## 14 ENSGALG00000034458             S100A1     100858258
## 15 ENSGALG00000010929            SPARCL1        422586
## 16 ENSGALG00000033226            FAM107B        426376
## 17 ENSGALG00000035317                ID3        395281</code></pre><pre class=r><code>summary(gene_ABmegre)</code></pre><pre><code>##  ensembl_gene_id    external_gene_name entrezgene_id      
##  Length:17          Length:17          Min.   :   373951  
##  Class :character   Class :character   1st Qu.:   395654  
##  Mode  :character   Mode  :character   Median :   422506  
##                                        Mean   : 29929910  
##                                        3rd Qu.:100533110  
##                                        Max.   :100859223</code></pre></div><div id=小结 class="section level3"><h3>小结</h3><p>学习使人懒惰，以前要花半小时进行EXCEL文件准备和ID转换，现在也就几分钟，成功偷懒~~</p></div><div id=参考资料 class="section level3"><h3><del>参考资料</del></h3></div></div><div class=article-tags><a class="badge badge-light" href=/tags/r-markdown/>R Markdown</a>
<a class="badge badge-light" href=/tags/academic/>Academic</a>
<a class="badge badge-light" href=/tags/rna-seq/>RNA-seq</a></div><div class="media author-card content-widget-hr"><img class="avatar mr-3 avatar-circle" src=/authors/admin/avatar_hua82a9dfdade94bdc4d0ff86ae39ca60b_179254_270x270_fill_q90_lanczos_center.jpg alt=Avatar><div class=media-body><h5 class=card-title><a href=/>Jiannan Zhang</a></h5><h6 class=card-subtitle>Postdoctoral Researcher</h6><p class=card-text>My research interests include Avian Physiology, Endocrinology and Metabolism.</p><ul class=network-icon aria-hidden=true><li><a href=/#contact><i class="fas fa-envelope"></i></a></li><li><a href=https://twitter.com/Jiannan_scu target=_blank rel=noopener><i class="fab fa-twitter"></i></a></li><li><a href="https://scholar.google.com/citations?user=qwOIhV4AAAAJ" target=_blank rel=noopener><i class="ai ai-google-scholar"></i></a></li><li><a href=https://github.com/biozhangjn target=_blank rel=noopener><i class="fab fa-github"></i></a></li></ul></div></div><div class="article-widget content-widget-hr"><h3>Related</h3><ul><li><a href=/post/bioconda/>Bioconda</a></li><li><a href=/post/shiny-server/>Shiny Server</a></li><li><a href=/post/blogdown-hugo-github-github-actions%E5%BB%BA%E7%AB%8B%E8%87%AA%E5%8A%A8%E6%8E%A8%E9%80%81blog/>Blogdown+Hugo+Github+Github Actions建立自动推送Blog</a></li><li><a href=/post/salmon/>Salmon代码备存</a></li><li><a href=/post/vscode-r/>VScode-R</a></li></ul></div></div></article><script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/highlight.min.js integrity="sha256-1zu+3BnLYV9LdiY85uXMzii3bdrkelyp37e0ZyTAQh0=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/languages/r.min.js></script><script>const code_highlighting=true;</script><script>const isSiteThemeDark=false;</script><script>const search_config={"indexURI":"/index.json","minLength":1,"threshold":0.3};const i18n={"no_results":"No results found","placeholder":"Search...","results":"results found"};const content_type={'post':"Posts",'project':"Projects",'publication':"Publications",'talk':"Talks"};</script><script id=search-hit-fuse-template type=text/x-template>
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script><script src=https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin=anonymous></script><script src=https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin=anonymous></script><script src=/js/academic.min.a8d7005002cb4a052fd6d721e83df9ba.js></script><div class=container><footer class=site-footer><p class=powered-by>Powered by the
<a href=https://sourcethemes.com/academic/ target=_blank rel=noopener>Academic theme</a> for
<a href=https://gohugo.io target=_blank rel=noopener>Hugo</a>.
<span class=float-right aria-hidden=true><a href=# class=back-to-top><span class=button_icon><i class="fas fa-chevron-up fa-2x"></i></span></a></span></p></footer></div><div id=modal class="modal fade" role=dialog><div class=modal-dialog><div class=modal-content><div class=modal-header><h5 class=modal-title>Cite</h5><button type=button class=close data-dismiss=modal aria-label=Close>
<span aria-hidden=true>&#215;</span></button></div><div class=modal-body><pre><code class="tex hljs"></code></pre></div><div class=modal-footer><a class="btn btn-outline-primary my-1 js-copy-cite" href=# target=_blank><i class="fas fa-copy"></i>Copy</a>
<a class="btn btn-outline-primary my-1 js-download-cite" href=# target=_blank><i class="fas fa-download"></i>Download</a><div id=modal-error></div></div></div></div></div></body></html>